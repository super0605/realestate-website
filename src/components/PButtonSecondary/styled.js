import styled from "styled-components";
import { Button, Typography } from "antd";

const { Text } = Typography;

export const CustomButton = styled(Button)`
  /* generel style */
  height: ${({ height }) => height};
  width: ${({ width }) => width};
  border: 0px;
  display: flex;
  padding: 0;
  align-items: center;
  background-color: transparent;
  box-shadow: unset;

  /* hover, focus, active style */
  &.ant-btn[disabled] {
    background-color: transparent;
    background: transparent;
    span {
      color: ${({ theme }) => theme.components.button.buttonSecondary.default.color};
      opacity: 0.4;
    }
  }
  &.ant-btn[disabled]:hover {
    background-color: transparent;
    span {
      color: ${({ theme }) => theme.components.button.buttonSecondary.default.color};
    }
  }
  &.ant-btn[disabled]:focus {
    span {
      color: ${({ theme }) => theme.components.button.buttonSecondary.default.color};
    }
  }

  &:hover {
    background-color: transparent;
    span {
      color: ${({ theme }) => theme.components.button.buttonSecondary.default.hoverColor};
    }
  }
`;

export const ButtonText = styled(Text)`
  ${({ theme }) => theme.components.button.buttonSecondary.textStyle}
  color: ${({ theme, ptype, pchecked }) =>
    pchecked === "true"
      ? theme.colors.primary
      : theme.components.button.buttonSecondary[ptype].color};
`;

export const SecondaryWrapper = styled(Text)`
  padding: 5px;
  ${({ theme }) => theme.components.button.buttonSecondary.textStyle}
  color: ${({ theme, ptype }) => theme.components.button.buttonSecondary[ptype].color};
`;
