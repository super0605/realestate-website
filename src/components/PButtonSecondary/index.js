import React, { useState } from "react";
import PropTypes from "prop-types";
import PCheckbox from "../PCheckbox";
import { CustomButton, ButtonText, SecondaryWrapper } from "./styled";

const PButtonSecondary = ({
  pname,
  ptype,
  psecondary,
  psecondaryAlign,
  pwithCheck,
  pchecked,
  width,
  height,
  children,
  ...rest
}) => {
  const [checked, setChecked] = useState(pchecked);
  const onClick = () => {
    pwithCheck && setChecked(!checked);
  };

  return (
    <CustomButton
      ptype={ptype}
      psecondary={psecondary}
      psecondaryalign={psecondaryAlign}
      pwithcheck={pwithCheck.toString()}
      pchecked={checked.toString()}
      height={height}
      width={width}
      onClick={onClick}
      {...rest}
    >
      {children ? (
        children
      ) : (
        <React.Fragment>
          {psecondaryAlign === "left" && (
            <SecondaryWrapper className="secondary-wrapper" ptype={ptype}>
              {!pwithCheck ? psecondary : <PCheckbox checked={checked} defaultChecked={pchecked} />}
            </SecondaryWrapper>
          )}
          <ButtonText className="button-text" ptype={ptype} pchecked={checked.toString()}>
            {pname}
          </ButtonText>
          {psecondaryAlign === "right" && (
            <SecondaryWrapper className="secondary-wrapper" ptype={ptype}>
              {!pwithCheck ? psecondary : <PCheckbox checked={checked} defaultChecked={pchecked} />}
            </SecondaryWrapper>
          )}
        </React.Fragment>
      )}
    </CustomButton>
  );
};

PButtonSecondary.propTypes = {
  pname: PropTypes.string,
  ptype: PropTypes.oneOf(["blue", "default"]),
  psecondary: PropTypes.node,
  psecondaryAlign: PropTypes.oneOf(["left", "right"]),
  pwithCheck: PropTypes.bool,
  pchecked: PropTypes.bool,
  width: PropTypes.string,
  height: PropTypes.string,
  children: PropTypes.node,
};

PButtonSecondary.defaultProps = {
  pname: "Button",
  ptype: "default",
  pwithCheck: false,
  pchecked: false,
  height: "24px",
};

export default PButtonSecondary;
