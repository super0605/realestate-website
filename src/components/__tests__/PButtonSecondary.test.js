import * as React from "react";
import { mount, shallow } from "enzyme";
import PButtonSecondary from "../PButtonSecondary";

describe("PButtonSecondary Component", () => {
  let props;
  let mountedPButtonSecondary;
  let wrapper;

  const renderWrapper = (bShallow) => {
    if (!mountedPButtonSecondary) {
      mountedPButtonSecondary = bShallow
        ? shallow(<PButtonSecondary {...props} />)
        : mount(<PButtonSecondary {...props} />);
    }
    return mountedPButtonSecondary;
  };

  beforeEach(() => {
    wrapper = renderWrapper(true);
  });

  it("should render without problems", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
