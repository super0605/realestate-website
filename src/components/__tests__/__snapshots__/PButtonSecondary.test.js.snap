// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PButtonSecondary Component should render without problems 1`] = `
<Styled(Button)
  height="24px"
  onClick={[Function]}
  pchecked="false"
  ptype="default"
  pwithcheck="false"
>
  <Styled(Text)
    className="button-text"
    pchecked="false"
    ptype="default"
  >
    Button
  </Styled(Text)>
</Styled(Button)>
`;
